config/opal_check_libnl.m4

```diff
@@ -0,0 +1,262 @@
+dnl -*- shell-script -*-
+dnl
+dnl Copyright (c) 2015-2016 Research Organization for Information Science
+dnl                         and Technology (RIST). All rights reserved.
+dnl $COPYRIGHT$
+dnl
+dnl Additional copyrights may follow
+dnl
+dnl $HEADER$
+dnl
+
+dnl
+dnl More libnl v1/v3 sadness: the two versions are not compatible
+dnl and will not work correctly if simultaneously linked into the
+dnl same applications.  Unfortunately, they *will* link into the
+dnl same image!  On platforms like SLES 12, libibverbs depends on
+dnl libnl-3.so.200 and friends, while a naive implementation of
+dnl our configure logic would link libnl.so.1 to libdaplusnic,
+dnl resulting in both versions in the dependency map at the same
+dnl time.
+dnl
+dnl GROSS: libnl wants us to either use pkg-config (which we
+dnl can't assume is always present) or we need to look in a
+dnl particular directory for the right libnl3 include files.  For
+dnl now, just hard code the special path into this logic.
+dnl
+dnl keep track of which libraries depend on libnl and which libraries
+dnl depend on libnl3. If Open MPI tries to link with libnl/libnl3, then
+dnl only consider the libnl version that is already depended on.
+dnl at the very end, abort is both libnl and libnl3 are required.
+
+dnl OPAL_LIBNL_SANITY_INIT()
+dnl --------------------------------------------------------------------
+AC_DEFUN([OPAL_LIBNL_SANITY_INIT], [
+    opal_libnl_version=0
+    opal_libnlv1_libs=
+    opal_libnlv3_libs=
+    AC_ARG_WITH([libnl],
+                [AC_HELP_STRING([--with-libnl(=DIR)],
+                                [Directory prefix for libnl (typically only necessary if libnl is installed in a location that the compiler/linker will not search by default)])],
+                [], [with_libnl=default])
+
+    # The --with options carry two pieces of information: 1) do
+    # you want a specific version of libnl, and 2) where that
+    # version of libnl lives.  For simplicity, let's separate
+    # those two pieces of information.
+    case "$with_libnl" in
+        no)
+            # Nope, don't want it
+            opal_want_libnl=no
+            ;;
+        yes)
+            # Yes, definitely want it
+            opal_want_libnl=yes
+            ;;
+        default)
+            # Default case -- try and see if we can find it
+            opal_want_libnl=default
+            opal_libnl_location=/usr
```

Comment:
If this /usr is just going to be stripped away in OPAL_CHECK_PACKAGE, then we probably shouldn't even list it here (i.e., leave it blank), because every time someone looks at this, they'll say "Oh, you shouldn't `-I/usr` or `-L/usr`, and spend time tracking down just to figure out that it's both useless and harmless.